tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/ExtraEntrySortedMap.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics.registry;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Ordering;
+import java.util.AbstractCollection;
+import java.util.AbstractMap;
+import java.util.AbstractSet;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+
+@SuppressWarnings({"JdkObsolete", "NullableDereference", "NullAway"})
+final class ExtraEntrySortedMap<K, V> extends AbstractMap<K, V> implements SortedMap<K, V> {
+    private final Ordering<? super K> ordering;
+    private final SortedMap<K, V> base;
+    private final K extraKey;
+    private final V extraValue;
+    private final int extraEntryHashCode;
+
+    ExtraEntrySortedMap(SortedMap<K, V> base, K extraKey, V extraValue) {
+        this.base = base;
+        this.extraKey = extraKey;
```

Comment:
seems like we assume `extraKey` and `extraValue` are non-null, can we assert?
```suggestion
        this.extraKey = Preconditions.checkNotNull(extraKey, "extraKey");
```